Lesson17/Ayse/contactBook.js

```diff
@@ -0,0 +1,331 @@
+  STEP 3: Add a New Contact
+-----------------------------------------------------------
+Function: addContact(name, phone, email)
+- Creates a new contact object and pushes it into 'contacts'.
+- Checks if a contact with the same name already
+  exists before adding. If found, logs a warning and returns.
+- Logs "Contact added successfully." if everything is good.
+*/
+console.log('STEP 3: Add a New Contact');
+
+
+let contacts3 = [
+  { name: 'Alice', phone: '123-456-7890', email: 'alice@gmail.com' },
+  { name: 'Anton', phone: '123-456-7899', email: 'anton@gmail.com' },
+  { name: 'Bob', phone: '222-222-2222', email: 'bob@gmail.com' }
+];
+
+function addContact(name, phone, email, contactList) {
+  for (let i = 0; i < contactList.length; i++) {
+    if (name === contactList[i].name) {
+      console.warn('The contact already exists');
+      return;
+    }
+  }
+  const contactListLengthPrev = contactList.length;
+  contactList.push({ name, phone, email });
+
+  const contactListLengthNew = contactList.length;
+
+  if (contactListLengthPrev < contactListLengthNew) {
+    console.log('Contact added successfully');
+  } else {
+    console.log('An error occurred when adding a contact');
+  }
+}
+
+  addContact("Charlie", "333-333-3333", "charlie@gmail.com", contacts3);
+  addContact("Alice", "111-111-1111", "alice_new@gmail.com", contacts3);
+  console.log(contacts3);
+
+
+/*
+-----------------------------------------------------------
+  STEP 4: View a Contact by Name
+-----------------------------------------------------------
+Function: viewContact(name)
+- Loops over 'contacts' to find one matching 'name'.
+- Logs the contact info if found.
+- Otherwise, logs: "No contact found with the name: <name>"
+*/
+
+console.log('STEP 4: View a Contact by Name');
+
+let contacts4 = [
+  { name: 'Alice', phone: '123-456-7890', email: 'alice@gmail.com' },
+  { name: 'Anton', phone: '123-456-7899', email: 'anton@gmail.com' },
+  { name: 'Bob', phone: '222-222-2222', email: 'bob@gmail.com' }
+];
+
+function viewContact(name, contactList) {
+  let isContactFound = false;
+  for (const contact of contactList) {
+    if (name === contact.name) {
+      console.log(
+        `name: ${contact.name}, phone:${contact.phone}, email:${contact.email}`,
+      );
+      isContactFound = true;
+      break;
+    }
+  }
+  if (!isContactFound) {
+    console.log(`No contact found with the name: ${name}`);
+  }
+}
+
+viewContact('Alice', contacts4);
+viewContact('Charlie', contacts4);
+
+/*
+-----------------------------------------------------------
+  STEP 5: Update a Contact
+-----------------------------------------------------------
+Function: updateContact(name, newPhone, newEmail)
+- Finds the contact by name and updates phone + email.
+- Logs "Contact updated successfully." if found.
+- Otherwise, logs: "No contact found with the name: <name>"
+*/
+console.log('STEP 5: Update a Contact');
+
+function findContact(name, contactList) {
```

Comment:
Move this function above all other functions that need to use it (otherwise you might be getting an error).
Modify the function to return contact or null so that you can use the contact directly.